tardisbase/testing/regression_comparison/visualize_files.py

```diff
@@ -0,0 +1,245 @@
+            except:
+                print(f"Warning: Could not access TARDIS repository at {self.tardis_repo_path}")
+
+        self.file_transitions = {}
+        self.all_files = set()
+        self.transition_columns = []
+
+
+    def get_h5_files_in_commit(self, commit_hash):
+        """
+        Extract all HDF5 files from a Git commit.
+
+        Parameters
+        ----------
+        commit_hash : str
+            Git commit hash to analyze.
+
+        Returns
+        -------
+        set of str
+            Set of .h5/.hdf5 file paths in the commit.
+        """
+        files = set()
+        try:
+            tree_output = self.repo.git.execute(['git', 'ls-tree', '-r', '--name-only', commit_hash])
+            for filepath in tree_output.split('\n'):
+                filepath = filepath.strip()
+                if filepath and filepath.endswith(('.h5', '.hdf5')):
+                    files.add(filepath)
+        except Exception as e:
+            print(f"Warning: Could not list files in commit {commit_hash[:8]}: {e}")
+        return files
+
+
+    def is_file_modified(self, file_path, older_commit, newer_commit):
+        """
+        Check if a file was modified between two commits.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to the file to check.
+        older_commit : str
+            Older commit hash.
+        newer_commit : str
+            Newer commit hash.
+
+        Returns
+        -------
+        bool
+            True if file was modified, False otherwise.
+        """
+        try:
+            diff_output = self.repo.git.diff(f'{older_commit}..{newer_commit}', '--', file_path)
+            return bool(diff_output.strip())
+        except Exception:
+            return False
+
+
+    def get_changes_with_git(self, older_commit, newer_commit):
+        """
+        Analyze file changes between two commits.
+
+        Parameters
+        ----------
+        older_commit : str
+            Older commit hash.
+        newer_commit : str
+            Newer commit hash.
+
+        Returns
+        -------
+        dict of str : str
+            File paths mapped to change symbols (A/D/M/•/−).
+        """
+        older_files = self.get_h5_files_in_commit(older_commit)
+        newer_files = self.get_h5_files_in_commit(newer_commit)
+        all_files = older_files | newer_files
+
+        changes = {}
+        for file_path in all_files:
+            in_older = file_path in older_files
+            in_newer = file_path in newer_files
+
+            if not in_older and not in_newer:
+                changes[file_path] = "−"
+            elif not in_older and in_newer:
+                changes[file_path] = "A"
+            elif in_older and not in_newer:
+                changes[file_path] = "D"
+            elif self.is_file_modified(file_path, older_commit, newer_commit):
+                changes[file_path] = "M"
+            else:
+                changes[file_path] = "•"
+
```

Comment:
is there a better way to do this? right now we are filtering by .h5 files. 
we want the filter perhaps for later use(more detailed comparisons) but for this case I dont think we should filter right?
can we do this for ALL the files instead of just HDF ones?